backend/routers/Pins.py

```
@@ -0,0 +1,126 @@
+from io import BytesIO
+from r2_storage import get_r2_storage
+
+load_dotenv()
+
+# --- Auth Helper Duplication Start ---
+# Duplicated from main.py to avoid circular imports and file creation restrictions.
+SECRET_KEY = os.getenv("SECRET_KEY")
+ALGORITHM = "HS256"
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="auth/login")
+
+def get_current_user(token: str = Depends(oauth2_scheme), db: Session = Depends(get_db)):
```

Ruff (0.14.8)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)